torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -12,13 +14,36 @@ using worker_id_t = int16_t;
 // A globally unique ID to identify an RpcAgent
 struct WorkerId {
   WorkerId(std::string name, int id)
-      : name_(std::move(name)), id_(id) {
+      : WorkerId(std::move(name), (worker_id_t)id) {
     TORCH_CHECK(id <= std::numeric_limits<worker_id_t>::max(),
         "RPC worker id ", id, " out of bound of int16_t.");
   }
 
   WorkerId(std::string name, worker_id_t id)
-      : name_(std::move(name)), id_(id) {}
+      : name_(std::move(name)), id_(id) {
+    bool validSize = name_.length() < MAX_NAME_LEN && name_.length() > 0;
+    bool validChar =
+        find_if(
+            name_.begin(),
+            name_.end(),
+            [](char c) {
+              return !(isalnum(c) || c == '-' || c == '_') ;
+            }
+        ) == name_.end();
+    TORCH_CHECK(validSize && validChar,
```

Comment:
Should we allow worker names to start with a digit, "-" or "_"? Seems like worker name should always start with an alphabet (usually this is what is done in other systems).

Reply:
I'd prefer not over engineering this here. What benefits do we or users get if we put more restrictions here? It looks fine to me if users want to do things like `init_model_parallel("1")` or `init_model_parallel("_hidden_monitor")`?

Reply:
I agree, let's keep this more flexible for now.